snuba/datasets/querylog.py

```diff
@@ -0,0 +1,78 @@
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+)
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.query.extensions import QueryExtension
+
+
+class QuerylogDataset(Dataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request_id", UUID()),
+                ("request_body", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("organization", Nullable(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
+                        [
+                            ("sql", String()),
+                            ("status", status_type),
+                            ("trace_id", Nullable(UUID())),
+                            ("duration_ms", UInt(32)),
+                            ("stats", String()),
```

Comment:
Actually after looking at commit https://github.com/getsentry/snuba/pull/783/commits/aee9d64c3a64c8db3ec2c77cb723618a8aa09f45, it does not seem bad to me. 
THere are a few fields we are going to query often. I would not remove them from the dataset as explicit columns whether you add or not a bare string with the whole stats.

Reply:
Fine, I can add it back if you think it's useful. Yes, I think we need the whole stats as well for ad hoc queries.

Reply:
Put those back now